src/main/java/org/reaktivity/nukleus/kafka/internal/cache/CompactedPartitionIndex.java

```diff
@@ -373,6 +376,7 @@ else if (invalidFrom != NO_POSITION)
 
     private long earliestOffset()
     {
+        compact();      // TODO keep track a validFromOffset
```

Comment:
Let's keep track of `validFromOffset` to avoid overhead of `compact()` on every call to `earliestOffset()` ?